pkg/master/storage_extensions.go

```diff
@@ -0,0 +1,112 @@
+package master
+
+import (
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/rest"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	extensionsapiv1beta1 "k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
+	"k8s.io/kubernetes/pkg/genericapiserver"
+	daemonetcd "k8s.io/kubernetes/pkg/registry/daemonset/etcd"
+	deploymentetcd "k8s.io/kubernetes/pkg/registry/deployment/etcd"
+	expcontrolleretcd "k8s.io/kubernetes/pkg/registry/experimental/controller/etcd"
+	horizontalpodautoscaleretcd "k8s.io/kubernetes/pkg/registry/horizontalpodautoscaler/etcd"
+	ingressetcd "k8s.io/kubernetes/pkg/registry/ingress/etcd"
+	jobetcd "k8s.io/kubernetes/pkg/registry/job/etcd"
+	networkpolicyetcd "k8s.io/kubernetes/pkg/registry/networkpolicy/etcd"
+	pspetcd "k8s.io/kubernetes/pkg/registry/podsecuritypolicy/etcd"
+	replicasetetcd "k8s.io/kubernetes/pkg/registry/replicaset/etcd"
+	thirdpartyresourceetcd "k8s.io/kubernetes/pkg/registry/thirdpartyresource/etcd"
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+type ExtensionsRESTStorageProvider struct {
+	ResourceInterface                     resourceInterface
```

Comment:
> Not really coming from this PR, but resourceInterface is really a bad name. Something like ThirdPartyRegistry describes better what it does.

I didn't even know this was a thing before.  It could use some love, but not in this pull.